app/uk/gov/hmrc/hec/services/DESService.scala

```diff
@@ -0,0 +1,105 @@
+/*
+ * Copyright 2021 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.hec.services
+
+import cats.data.EitherT
+import cats.implicits._
+import com.google.inject.{ImplementedBy, Inject, Singleton}
+import play.api.http.Status.{BAD_REQUEST, NOT_FOUND, OK}
+import play.api.libs.json.{Json, Reads}
+import uk.gov.hmrc.hec.connectors.DESConnector
+import uk.gov.hmrc.hec.models.Error
+import uk.gov.hmrc.hec.models.ids.{CRN, CTUTR}
+import uk.gov.hmrc.hec.services.DESService.{BackendError, DESError, DataNotFoundError, InvalidCRNError}
+import uk.gov.hmrc.hec.services.DESServiceImpl._
+import uk.gov.hmrc.hec.util.HttpResponseOps._
+import uk.gov.hmrc.hec.util.Logging
+import uk.gov.hmrc.http.{HeaderCarrier, HttpResponse}
+
+import scala.concurrent.{ExecutionContext, Future}
+
+@ImplementedBy(classOf[DESServiceImpl])
+trait DESService {
+
+  def getCtutr(crn: CRN)(implicit hc: HeaderCarrier): EitherT[Future, DESError, CTUTR]
+
+}
+
+object DESService {
+  sealed trait DESError extends Product with Serializable
```

Comment:
shall we have msg here as well like DESError(msg: String) and other case objects are overriding it, so that it's easy to write it's implicit Writes in hec-stubs?

Reply:
`msg` is not common to all the case classes that extend it, so can't put it in `DESError`

i think in `hec-stubs` we can just use json for errors just like we do for the success responses